File: test/templates.test.js
import test from "node:test";
import assert from "node:assert/strict";
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";
import yaml from "js-yaml";
import { validateManifest, validateNote } from "../brain_tools/schema.js";
import { read } from "../brain_tools/frontmatter.js";

const TEMPLATES = join(dirname(fileURLToPath(import.meta.url)), "..", "templates");

test("area template is a valid manifest", () => {
  const data = yaml.load(readFileSync(join(TEMPLATES, "area.brain.yml"), "utf-8"));
  assert.deepEqual(validateManifest(data), []);
});

test("note template is a valid note", () => {
  const { meta } = read(join(TEMPLATES, "note.md"));
  assert.deepEqual(validateNote(meta), []);
});

File: src/test/java/brain/TemplatesTest.java
package brain;

import static org.junit.jupiter.api.Assertions.assertEquals;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;
import java.util.Map;
import org.junit.jupiter.api.Test;

class TemplatesTest {
  @Test
  void areaTemplateIsValidManifest() throws Exception {
    Map<String, Object> data = Yamls.load(Files.readString(Path.of("templates/area.brain.yml")));
    assertEquals(List.of(), Schema.validateManifest(data));
  }

  @Test
  void noteTemplateIsValidNote() throws Exception {
    Map<String, Object> meta = Frontmatter.read(Path.of("templates/note.md")).meta();
    assertEquals(List.of(), Schema.validateNote(meta));
  }
}
